src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1028,42 +1030,127 @@ public int deleteInactiveConsumers(Collection<String> inactiveConsumerIds) {
+        }
+
+        String retrieveScaCertsSql = "SELECT cont_acc_cert_id FROM cp_consumer WHERE id IN (:consumerIds)";
+        return this.currentSession()
+            .createSQLQuery(retrieveScaCertsSql)
+            .setParameter("consumerIds", consumerIds)
+            .getResultList();
+    }
+
+    /**
+     * Retrieves all the serial ids for provided {@link ContentAccessCertificate} ids
+     * and {@link IdentityCertificate} ids.
+     *
+     * @param caCertIds - ids of content access certificates.
+     * @param idCertIds - ids of identity certificates.
+     * @return serial ids.
+     */
+    public List<Long> getSerialIdsForCerts(Collection<String> caCertIds, Collection<String> idCertIds) {
+        List<BigInteger> serialIds = new ArrayList<>();
+        if (caCertIds != null && !caCertIds.isEmpty()) {
+            String caCertSql =
+                "SELECT ca.serial_id AS serialIds " +
+                "FROM cp_cont_access_cert ca " +
+                "WHERE ca.id IN (:certIds)";
 
-                this.delete(consumer);
-            });
+            serialIds.addAll(this.currentSession()
+                .createSQLQuery(caCertSql)
+                .setParameter("certIds", caCertIds)
+                .getResultList());
+        }
 
-        // Need to flush to avoid foreign key constraint violations with the removed consumers.
-        flush();
+        if (idCertIds != null && !idCertIds.isEmpty()) {
+            String idCertSql =
+                "SELECT idcert.serial_id AS serialIds " +
+                "FROM cp_id_cert idcert " +
+                "WHERE idcert.id IN (:certIds)";
 
-        int deletedIdentityCerts = this.identityCertificateCurator.deleteByIds(idCertsToDelete);
-        log.debug("Deleted {} identity certificates", deletedIdentityCerts);
+            serialIds.addAll(this.currentSession()
+                .createSQLQuery(idCertSql)
+                .setParameter("certIds", idCertIds)
+                .getResultList());
+        }
 
-        int deletedContentAccessCerts = this.contentAccessCertificateCurator.deleteByIds(caCertsToDelete);
-        log.debug("Deleted {} content access certificates", deletedContentAccessCerts);
+        return serialIds.stream().map(BigInteger::longValue).collect(Collectors.toList());
+    }
 
-        int revokedSerials = this.certificateSerialCurator.revokeByIds(serialsToRevoke);
-        log.debug("Revoked {} certificate serials", revokedSerials);
+    /**
+     * Deletes {@link Consumer}s based on the provided consumer ids.
+     *
+     * @param consumerIds - ids of the consumers to delete.
+     * @return the number of consumer that were deleted.
+     */
+    public int deleteConsumers(Collection<String> consumerIds) {
+        if (consumerIds == null || consumerIds.isEmpty()) {
+            return 0;
+        }
 
-        return inactiveConsumerIds.size();
+        String deleteConsumersStatement = "DELETE FROM cp_consumer WHERE id IN (:consumerIds)";
```

Comment:
Changed to use HQL.

Reply:
Looks like the other ones got updated, but the query for this comment in particular still looks to be native.

Reply:
Oops, yeah I missed this one. I removed the delete statement based on the inactive consumer ids because of the constraint violations. I added a comment in regards to that!